.tests/tests.sh

```diff
@@ -4,10 +4,14 @@ set -ve
 
 GROUP_VARS=src/commcare_cloud/ansible/group_vars/all.yml
 eval PYVER=$(grep python_version: $GROUP_VARS | head -n1 | cut -d' ' -f2)
-# Link Travis-installed python where ansible virtualenv will find it
+# Link GitHub Actions-installed python where ansible virtualenv will find it
 [ -z $(which python${PYVER}) ] && sudo ln -s /opt/python/${PYVER}/bin/*${PYVER}* /usr/local/bin/
 which python${PYVER}
 python${PYVER} --version
+# link virtualenv to a place where it will be found in ansible
+which /opt/pipx_bin/virtualenv && sudo ln -s /opt/pipx_bin/virtualenv /usr/local/bin/
+which virtualenv
+virtualenv --version
```

Comment:
I'm confused by the use of `which` here.
In the first case a shell test seems more appropriate.
The second one appears to be useful for debugging, but the result should never vary.
Unless the idea was to perform a check similar to the previous one for `python${PYVER}`?

Reply:
Yeah the lines
```
which virtualenv
virtualenv --version
```
are just for debugging. I saw that we did that for python a few lines above and I thought it seemed useful.

Reply:
Then what about the same approach?

```suggestion
# link virtualenv to a place where it will be found in ansible
[ -z $(which virtualenv) ] && sudo ln -s /opt/pipx_bin/virtualenv /usr/local/bin/
which virtualenv
virtualenv --version
```

Reply:
That may not work because `/opt/pipx_bin/` isn't necessarily on the $PATH the way it's set up. That's why I was using the full path to the executable.

So I can learn, what's the benefit here of `[ -z $(...)] &&` over just `... &&`? Does it have nicer behavior like suppressing the output or something, or is it just for readability i.e. easier to tell I'm using it as a condition?

Reply:
In the suggested change `/opt/pipx_bin/virtualenv` is linked to `/usr/local/bin/` only if a `virtualenv` is not already present in the PATH (same as what the block before does for `python${PYVER}`).
If this is unnecessary and you only want to check for the existence of `/opt/pipx_bin/virtualenv` before linking it, then I think `[ -e /opt/pipx_bin/virtualenv ] && ...` would be preferable, because, yes, it's a more standard/readable approach and won't produce the redundant output from `which`.
But in this case is the check necessary at all? Is there a scenario in which `/opt/pipx_bin/virtualenv` wouldn't be available?

Reply:
I think the benefit I was trying to get out of using `which /opt/pipx_bin/virtualenv && ` is that I wanted this line to fail hard (cause tests to halt and report error) if this was not found. This could happen if we change something about the corresponding setup or if a default behavior changes out from under us over time. That would let us get the error here rather than much later when we try to use `virtualenv` but `/usr/local/bin/virtualenv` is a broken link.

I think `[ -e /opt/pipx_bin/virtualenv ] && ` will have that same behavior (failing hard if that file is not an executable). So I like the suggestion and I'll use that.